src/tools/asm2wasm.cpp

```diff
@@ -200,7 +207,11 @@ int main(int argc, const char *argv[]) {
   writer.setDebugInfo(passOptions.debugInfo);
   writer.setSymbolMap(symbolMap);
   writer.setBinary(emitBinary);
-  writer.write(wasm, options.extra["output"]);
+  if (emitBinary && binaryMapFile.size()) {
+    writer.writeBinary(wasm, options.extra["output"], binaryMapFile, binaryMapUrl);
```

Comment:
i think a more consistent api might be to have, like we have `setBinary` now (used a few lines above), `setBinaryMapFile` and `setBinaryMapURL` methods that we call. then we can avoid calling `writeBinary` here.